text/0000-no-more-extern-crate.md

```diff
@@ -0,0 +1,136 @@
+will no longer have to specify `extern crate`:
+dependencies added to `Cargo.toml` will be automatically imported.
+Projects which require more flexibility can still use manual `extern crate`
+and will be unaffected by this RFC.
+
+# Motivation
+[motivation]: #motivation
+
+One of the principles of Rust is that using external crates should be as
+easy and natural as using the standard library.
+This allows the standard library to be kept small, and allows mature, standard
+solutions to be developed by the community.
+
+Currently, however, external crates must be specified twice: once in a build
+system such as Cargo and again in the source code using `extern crate`.
+When external dependencies are conditional (`cfg`) upon feature flags or the
+target platform, the conditional logic must appear in both `Cargo.toml` and
+the `extern crate` declarations.
+
+This duplication causes unnecessary effort and results in one more opportunity
+for mistakes when working with conditionally-enabled dependencies.
+Allowing the omission of the redundant `extern crate` syntax contributes to the
+roadmap goals of
+[improving Rust's ergonomics](https://github.com/rust-lang/rust-roadmap/issues/17)
+and
+[providing easy access to high-quality crates.](https://github.com/rust-lang/rust-roadmap/issues/9)
+
+# Guide-Level Explanation
+[guide]: #guide
+
+When you add a dependency to your `Cargo.toml`, it is immediately usable within
+the source of your crate:
+
+```toml
+# Cargo.toml:
+name = "my_crate"
+version = "0.1.0"
+authors = ["Me" <me@mail.com>]
+
+[dependencies]
+rand = "0.3"
+```
+
+```rust
+// src/main.rs:
+
+fn main() {
+    println!"A random character: {}", rand::random::<char>());
+}
+```
+
+# Reference-Level Explanation
+[reference]: #reference
+
+External crates can be passed to the rust compiler using the
+`--extern CRATE_NAME=PATH` flag.
+For example, `cargo build`-ing a crate `my_crate` with a dependency on `rand`
+results in a call to rustc that looks something like
+`rustc --crate-name mycrate src/main.rs --extern rand=/path/to/librand.rlib ...`.
+
+When an external crate is specified this way,
+the crate will automatically brought into scope as if an
+`extern crate name_of_crate;`
+declaration had been added to the current crate root.
+This behavior won't occur when including a library using the `-l`
+or `-L` flags.
+
+We will continue to support the current `extern crate` syntax,
+both for backwards compatibility and to enable users who want to use manual
+`extern crate` in order to have more fine grained control-- say, if they wanted
+to import an external crate only inside an inner module.
+No automatic import will occur if an `extern crate` declaration for the same
+external dependency appears anywhere within the crate.
+For example, if `rand = "0.3"` is listed as a dependency in Cargo.toml
+and `extern crate rand;` appears somewhere in the crate being compiled,
+then no implicit `extern crate rand;` will be added.
```

Comment:
The motivation for `extern crate` anywhere disabling the feature is just about not polluting namespace. We've already established that the RFC shouldn't result in automatic inclusion in local namespace. It can possibly be further specified that the automatic path links can be shadowed by root modules (which would make them only accessible via `extern crate`, but fixes backcompat).